packages/truffle-compile/test/test_supplier.js

```diff
@@ -144,45 +135,35 @@ describe("CompilerSupplier", function() {
       const cachedOptions = Config.default().merge(options);
 
       // Run compiler, expecting solc to be downloaded and cached.
-      compile(version4PragmaSource, cachedOptions, err => {
-        if (err) return done(err);
-
-        assert(fs.existsSync(expectedCache), "Should have cached compiler");
-
-        // Get cached solc access time
-        initialAccessTime = fs.statSync(expectedCache).atime.getTime();
-
-        // Wait a second and recompile, verifying that the cached solc
-        // got accessed / ran ok.
-        waitSecond()
-          .then(() => {
-            compile(version4PragmaSource, cachedOptions, (err, result) => {
-              if (err) return done(err);
-
-              finalAccessTime = fs.statSync(expectedCache).atime.getTime();
-
-              assert(
-                result["NewPragma"].contract_name === "NewPragma",
-                "Should have compiled"
-              );
-
-              // atime is not getting updatd on read in CI.
-              if (!process.env.TEST) {
-                assert(
-                  initialAccessTime < finalAccessTime,
-                  "Should have used cached compiler"
-                );
-              }
+      await compile(version4PragmaSource, cachedOptions);
 
-              done();
-            });
-          })
-          .catch(done);
-      });
+      assert(fs.existsSync(expectedCache), "Should have cached compiler");
```

Comment:
Is there a reason to use `fs.existsSync()` over `fs.exists()` or `fse.exists()`? Looks like there's no problem awaiting here so may be worthwhile removing the synchronous function?

Reply:
Ah that's just legacy code, I didn't add that. I'll make the change.